service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java

```diff
@@ -80,59 +79,122 @@ public String toString() {
      *
      * @param key
      *            The Policy Evaluation key to retrieve.
-     * @return The Policy Evaluation Result if the key is the the cache and the result isn't invalidated, or null
+     * @return The Policy Evaluation Result if the key is in the cache and the result isn't invalidated, or null
      */
     @Override
     public PolicyEvaluationResult get(final PolicyEvaluationRequestCacheKey key) {
-        String redisKey = key.toRedisKey();
-        List<String> keys = assembleKeys(key);
-        List<String> values = multiGet(keys);
+        //Get all result related entries
+        DecisionCacheEntries cachedEntries = new DecisionCacheEntries(key);
 
-        logCacheGetDebugMessages(key, redisKey, keys, values);
-        if (!isRequestCached(values)) {
+        String cachedEvalResultString = cachedEntries.getDecisionString();
+        if (null == cachedEvalResultString) {
             return null;
         }
+        PolicyEvaluationResult cachedEvalResult = toPolicyEvaluationResult(cachedEvalResultString);
+        
+        List<String> invalidationTimeStamps = new ArrayList<>();
+        invalidationTimeStamps.add(cachedEntries.getSubjectLastModified());
+        invalidationTimeStamps.addAll(cachedEntries.getPolicySetsLastModified());
+
+        Set<String> cachedResolvedResourceUris = cachedEvalResult.getResolvedResourceUris();
+        //is requested resource id same as resolved resource uri ?
+        if (cachedResolvedResourceUris.size() == 1
+                && cachedResolvedResourceUris.toArray()[0].equals(key.getResourceId())) {
```

Comment:
Use cachedResolvedResourceUris.iterator().next() vs toArray() so that we don't have to copy contents to array.